src/stream/Encryption.ts

```diff
@@ -239,29 +312,57 @@ class EncryptionUtilBase {
             streamMessage.serializedContent = serializedContent
         } catch (err) {
             streamMessage.encryptionType = StreamMessage.ENCRYPTION_TYPES.AES
-            throw new UnableToDecryptError(err.message, streamMessage)
+            throw new UnableToDecryptError(err.stack, streamMessage)
+        }
+
+        try {
+            const { newGroupKey } = streamMessage
+            if (newGroupKey) {
+                // newGroupKey should be EncryptedGroupKey | GroupKey, but GroupKey is not defined in protocol
+                // @ts-expect-error
+                streamMessage.newGroupKey = GroupKey.from([
+                    newGroupKey.groupKeyId,
+                    this.decrypt(newGroupKey.encryptedGroupKeyHex, groupKey)
+                ])
+            }
+        } catch (err) {
+            streamMessage.encryptionType = StreamMessage.ENCRYPTION_TYPES.AES
+            throw new UnableToDecryptError('Could not decrypt new group key: ' + err.stack, streamMessage)
         }
         return null
         /* eslint-enable no-param-reassign */
     }
 }
 
+// after EncryptionUtil is ready
+type InitializedEncryptionUtil = O.Overwrite<EncryptionUtil, {
```

Comment:
Is this used for security reasons? Should we use the same kind of construct for private key authentication in `StreamrClientOptions`?

Reply:
Yes I was thinking exactly that.

Reply:
Not for "security" but more to help the typing.